pkg/api/admin_users.go

```diff
@@ -176,6 +176,10 @@ func (hs *HTTPServer) AdminUpdateUserPassword(c *contextmodel.ReqContext) respon
 // 403: forbiddenError
 // 500: internalServerError
 func (hs *HTTPServer) AdminUpdateUserPermissions(c *contextmodel.ReqContext) response.Response {
+	if !c.SignedInUser.IsGrafanaAdmin {
+		return response.Error(http.StatusForbidden, "You must be a Grafana Admin to grant and revoke Grafana Admin permissions", nil)
+	}
```

Comment:
An alternative would be to use [this](https://github.com/grafana/grafana/blob/8f0f0387b8d316d73a279e30d6ad264070047441/pkg/api/api.go#L57) middleware for this endpoint. Either way is fine

Reply:
Ah nice, I forgot about that one. I'll use the middleware then to slightly centralise the access checks.